File: javascript/3637.Trionic-Array-I.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isTrionic = function (nums) {
  const n = nums.length;
  let p = -1;
  let q = -1;
  let i = 0;

  if (n < 4) return false;

  while (i + 1 < n && nums[i + 1] > nums[i]) {
    i += 1;
  }

  p = i;

  while (i + 1 < n && nums[i + 1] < nums[i]) {
    i += 1;
  }

  q = i;

  while (i + 1 < n && nums[i + 1] > nums[i]) {
    i += 1;
  }

  return p > 0 && p < q && q < n - 1 && i === n - 1;
};

File: python/3637.Trionic-Array-I.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        n = len(nums)
        i = 0
        p = -1
        q = -1

        if n < 4:
            return False

        while i + 1 < n and nums[i + 1] > nums[i]:
            i += 1
        p = i

        while i + 1 < n and nums[i + 1] < nums[i]:
            i += 1
        q = i

        while i + 1 < n and nums[i + 1] > nums[i]:
            i += 1

        return p > 0 and p < q and q < n - 1 and i == n - 1